messaging/src/main/java/org/axonframework/eventhandling/deadletter/DeadLetteringEventHandlerInvoker.java

```diff
@@ -0,0 +1,286 @@
+    }
+
+    /**
+     * Instantiate a builder to construct a {@link DeadLetteringEventHandlerInvoker}.
+     * <p>
+     * The {@link ListenerInvocationErrorHandler} is defaulted to a {@link PropagatingErrorHandler}, the
+     * {@link SequencingPolicy} to a {@link SequentialPerAggregatePolicy}, and {@code allowReset} defaults to
+     * {@code false}. Providing at least one Event Handler, a {@link DeadLetterQueue}, a {@code processingGroup} name,
+     * and a {@link TransactionManager} are <b>hard requirements</b> and as such should be provided.
+     *
+     * @return A builder that can construct a {@link DeadLetteringEventHandlerInvoker}.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public void handle(@Nonnull EventMessage<?> message, @Nonnull Segment segment) throws Exception {
+        if (!super.sequencingPolicyMatchesSegment(message, segment)) {
+            logger.trace("Ignoring event [{}] as it is not meant for segment [{}].", message, segment);
+            return;
+        }
+
+        EventHandlingQueueIdentifier identifier =
+                new EventHandlingQueueIdentifier(super.sequenceIdentifier(message), processingGroup);
+        Optional<DeadLetter<EventMessage<?>>> optionalLetter =
+                transactionManager.fetchInTransaction(() -> queue.enqueueIfPresent(identifier, message));
+        if (optionalLetter.isPresent()) {
+            if (logger.isInfoEnabled()) {
+                logger.info("Event [{}] is added to the dead-letter queue since its queue id [{}] is already present.",
+                            message, identifier.combinedIdentifier());
+            }
+        } else {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Event [{}] with queue id [{}] is not present in the dead-letter queue present."
+                                     + "Handle operation is delegated to the wrapped EventHandlerInvoker.",
+                             message, identifier.combinedIdentifier());
+            }
+            try {
+                super.invokeHandlers(message);
+            } catch (Exception e) {
+                transactionManager.executeInTransaction(() -> queue.enqueue(identifier, message, e));
+            }
+        }
+    }
+
+    @Override
+    public void performReset() {
+        if (allowReset) {
+            transactionManager.executeInTransaction(() -> queue.clear(processingGroup));
+        }
+        super.performReset(null);
+    }
+
+    @Override
+    public <R> void performReset(R resetContext) {
+        if (allowReset) {
+            transactionManager.executeInTransaction(() -> queue.clear(processingGroup));
+        }
+        super.performReset(resetContext);
+    }
+
+    @Override
+    public void registerLifecycleHandlers(LifecycleRegistry lifecycle) {
+        lifecycle.onStart(Phase.INBOUND_EVENT_CONNECTORS + 1, this::start);
```

Comment:
The shutdown is performed within the DLQ, as that runs the evaluation tasks for enqueued message. 

Mainly because a DLQ _might_ life without this invoker. 

Hence, it didn't feel reasonable to me to have the invoker shut down the DLQ.

Reply:
As the `DeadLetterQueue` nor the `DeadLetteringEventHandlerInvoker` has an `Executor`, there's nothing to start or stop through the Configuration anymore. 
Hence, the lifecycle handlers have been removed from this pull request.